### PR TITLE
SEC: fix security breaches in GHA workflows detected with zizmor

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
This fixes a couple security issues I found using [`zizmor`](https://github.com/woodruffw/zizmor).

One more issue remains unadressed because I do not understand what the problem is or how to address it. Here's the current output:

```
❯ zizmor .github/workflows
🌈 completed ci_cron_daily.yml
🌈 completed check_milestone.yml
🌈 completed publish.yml
🌈 completed ci_benchmark.yml
🌈 completed codeql-analysis.yml
🌈 completed update_astropy_iers_data_pin.yml
🌈 completed stalebot.yml
🌈 completed ci_cron_weekly.yml
🌈 completed open_actions.yml
🌈 completed CFF-test.yml
🌈 completed check_changelog.yml
🌈 completed ci_workflows.yml
error[dangerous-triggers]: use of fundamentally insecure workflow trigger
 --> /Users/clm/dev/astropy-project/coordinated/astropy/.github/workflows/open_actions.yml:3:1
  |
3 | / on:
4 | |   issues:
... |
8 | |     types:
9 | |     - opened
  | |____________^ pull_request_target is almost always used insecurely
  |

1 findings (0 unknown, 0 informational, 0 low, 0 medium, 1 high)
```

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
> error[dangerous-triggers]

Well, not like I did it for fun. We need extra access to post comment on PR. Will the zizmor dev able to give advice?

File: .github/workflows/update_astropy_iers_data_pin.yml
Comment (pllim) [inline .github/workflows/update_astropy_iers_data_pin.yml:1]: Not really a way to check this on PR but I can probably do a trigger after merge if necessary.

**Comment by neutrinoceros** [comment]
Apparently he's very reactive yes. So I heard.

**Comment by WilliamJamieson** [comment]
Do we want to add this to the pre-commit? https://github.com/woodruffw/zizmor/blob/a3e84e6564e6025ce4c5e7bb3ab024d5dbf62901/docs/usage.md?plain=1#L106-L130

**Comment by pllim** [comment]
> pre-commit

Is that necessary? We do not add new workflows often.

**Comment by pllim** [comment]
> very reactive yes

Is that good or bad? 😅

**Comment by neutrinoceros** [comment]
> Do we want to add this to the pre-commit?

oooh I didn't see there was a pre-commit hook already.

> Is that necessary? We do not add new workflows often.

Sure, but zizmor might add new checks in the future (the tool is extremely new), that we might want to know about sooner rather than later.

> Is that good or bad? 

good !

**Comment by neutrinoceros** [comment]
[zizmor 0.6](https://github.com/woodruffw/zizmor/releases/tag/v0.6.0) now supports `# zizmor: ignore[rule]` comments, so, we could leverage it to ignore the one `dangerous-triggers` violation we have and add the pre-commit hook to catch future breaches !

**Comment by pllim** [comment]
I guess doesn't hurt...

**Review by WilliamJamieson — COMMENTED** [review]
This seems reasonable to me.

**Comment by neutrinoceros** [comment]
pre-commit.ci is having issues building zizmor 0.6.0 (and I have to confess I did to; I needed to update `rust` on my system), so I guess we should hold on that pre-commit hook...

**Review by pllim — APPROVED** [review]
Thanks!

If you want, feel free to open follow up issue for the pre-commit.

**Comment by neutrinoceros** [comment]
I'll try a manual backport tomorrow !

**Pull request merged**